bin/node/cli/tests/websocket_server.rs

```diff
@@ -16,7 +16,8 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-use async_std::net::{TcpListener, TcpStream};
+use tokio::net::{TcpListener, TcpStream};
+use tokio_util::compat::{Compat, TokioAsyncReadCompatExt};
```

Comment:
Why do need compatibility mode?

Reply:
This is because soketto WebSocket server expects `TcpStream` to be `futures::io::AsyncRead` and not `tokio::io::AsyncRead`, so we need to wrap tokio's `TcpStream` with a `.compat()` call here: https://github.com/paritytech/substrate/blob/398d10c3ee6d8f0c48acbb0f81401d93fb25245f/bin/node/cli/tests/websocket_server.rs#L128

Surprisingly, I also haven't found other alternative with libp2p protocol upgrades, which also expect `futures::io::AsyncRead`, so tokio's `TcpStream`s are also wrapped in `.compat()` there. Please correct me if there is a better way around. https://github.com/paritytech/substrate/blob/398d10c3ee6d8f0c48acbb0f81401d93fb25245f/client/network/src/protocol/notifications/upgrade/notifications.rs#L501-L505

Reply:
Ty for the explanation! I should have checked before. I thought that `Compat` is for making it compatible for older tokio versions. There was once a similar construct for using futures v0.3 and v0.1 together ;) That confused me here. But it sounds good what you are doing here!